Routers/BookingRouter.cs

```diff
@@ -0,0 +1,118 @@
+﻿using AirportTicketBooking.Classes;
+using AirportTicketBooking.Repositories;
+using ClassType = TicketClassType.ClassType;
+
+namespace AirportTicketBooking.Routers
+{
+    internal class BookingRouter
+    {
+        private BookingRepositoty _bookingRepositoty;
+        private FlightRouter _flightRouter;
+
+        public BookingRouter()
+        {
+            _bookingRepositoty = new BookingRepositoty();
+        }
+
+        public List<Booking> GetAllData()
+        {
+            return _bookingRepositoty.Load();
+        }
+
+        public List<Booking> GetAll()
+        {
+            return GetAllData().Where(booking => booking.ValidationMessage == null).ToList();
+        }
+
+        public bool TryGetByBookingId(int bookingId, out Booking booking)
+        {
+            booking = GetAll().FirstOrDefault(booking => booking.BookingId == bookingId);
+            return (booking != null);
+        }
+
+        public void Add(Booking booking)
+        {
+            _bookingRepositoty.Save(booking);
+        }
+
+        public void RemoveByBookingId(int bookingId)
+        {
+            if (!TryGetByBookingId(bookingId, out _))
+            {
+                throw new Exception($"Booking with BookingId: {bookingId} doesn't exist");
+            }
+
+            var remainingBookings = GetAll().Where(booking => booking.BookingId != bookingId).ToList();
+
+            _bookingRepositoty.DeleteTheFileData();
+
+            foreach (var booking in remainingBookings)
+            {
+                _bookingRepositoty.Save(booking);
+            }
```

Comment:
Delete all the records for the sake of deleting one record and then re-inserting the records is a good idea. 

I believe you can find a way to just delete the record you are interested in by knowing the index of the line for that record in the file and then clearing that line only.

Reply:
For this, I was thinking of deleting a specific line in the file, but I don't know how to code it, I thought about overwriting the line with ( \n ) or like what I did. Is there a better way?